### DISCUSSION TITLE
Testing select compoments with testing library

### DISCUSSION THREAD

**Discussion by rogix** [mantinedev/mantine#1388] [Q&A] [answered]
Great lib! 

I was wondering what approach you people use to test Select component. Take this as an example:

```javascript
   <Select
      label="Your favorite framework/library"
      placeholder="Pick one"
      data={[
        { value: 'react', label: 'React' },
        { value: 'ng', label: 'Angular' },
        { value: 'svelte', label: 'Svelte' },
        { value: 'vue', label: 'Vue' },
      ]}
    />
```

I would like to change or check which option was selected. I could do something like: 

```javascript
userEvent.selectOptions(selectComponent, 'react')
```

to update the value and test against it, if it was a regular `select/option` api.

But this is not the case. 

React-select has a utility library to help with this https://github.com/romgain/react-select-event.

Is there a straight way to test this king of component in mantine? Any help would be welcome, thanks!

**Comment by rtivital** [comment, marked as answer]
Currently, as far as I know, there is no way to select value with userEvent, and we cannot provide any examples from our side, except those that are within tests in this repo.

**Reply by rogix** [reply]
Thanks for the reply, @rtivital. I agree. Probably is not possible to test it this way. 

The problem I am facing is that I have other components state, like buttons state and other dynamics select field that depends on the value selected. 

For example if I select `React` the button should be active, else it should be disabled and so on. I am not finding a way to test this. Anyway, I will keep looking. Thanks! 

... and thanks for the lib. It's a great lib
